pkg/agent/route/route_linux.go

```diff
@@ -431,6 +455,24 @@ func (c *Client) restoreIptablesData(podCIDR *net.IPNet, podIPSet string, snatMa
 		"-j", iptables.MasqueradeTarget, "--random-fully",
 	}...)
 
+	if c.proxyFull {
+		if podCIDR.IP.To4() != nil {
+			writeLine(iptablesData, []string{
+				"-A", antreaPostRoutingChain,
+				"-m", "comment", "--comment", `"Antrea: masquerade Service host network Endpoint traffic"`,
+				"-s", serviceGWHairpinIPv4.String(),
```

Comment:
I actually feel bad route code needs to understand AntreaProxy logics and things like Service hairpin IP. Ideally we should isolate the logics, to avoid many components need to understand a feature.

Maybe we can say Route Client and OpenflowClient are special and need have feature specific code. But do you have some ideas? Also add @tnqn and @wenyingd to see what they think.